examples/cms/src/App.tsx

```diff
@@ -0,0 +1,87 @@
+import "./App.css"
+
+import type { ManagedCollection } from "framer-plugin"
+
+import { framer } from "framer-plugin"
+import { useEffect, useLayoutEffect, useState } from "react"
+import { DataSource, getDataSource, getDataSourcesIds } from "./data"
+import { FieldMapping } from "./FieldMapping"
+import { SelectDataSource } from "./SelectDataSource"
+import { Spinner } from "./components/Spinner"
+
+interface AppProps {
+    collection: ManagedCollection
+    previouslyConfiguredDataSourceId: string | null
+    previouslyConfiguredSlugFieldId: string | null
+}
+
+export function App({ collection, previouslyConfiguredDataSourceId, previouslyConfiguredSlugFieldId }: AppProps) {
+    const [dataSource, setDataSource] = useState<DataSource | null>(null)
+
+    const [isLoadingConfiguredDataSource, setIsLoadingConfiguredDataSource] = useState(
```

Comment:
do we need to say "configured" each time, some of these lines become really long with these variable names and its harder to read